Reorient tracks according to ROI

arokem opened the pull request:
This is important in order to correctly pull out a core of a bundle based on a median operation on the coordinates. For details, see: 

https://github.com/arokem/AFQ-notebooks/blob/08ac2fc66d1ded0234eec898438d0c7f8380cafc/AFQ-registration-callosum.ipynb

jchoude commented on dipy/tracking/streamline.py:
```diff
@@ -251,3 +251,63 @@ def select_by_rois(streamlines, rois, include, mode=None, affine=None,
+    Parameters
+    ----------
+    streamlines : list
+        List of 3d arrays. Coordinates of the
```
Missing the end of the comment

jchoude commented on dipy/tracking/streamline.py:
```diff
@@ -251,3 +251,63 @@ def select_by_rois(streamlines, rois, include, mode=None, affine=None,
+        Binary masks designating the location of the regions of interest, or
+        coordinate arrays (3d coordinates)
+    affine : ndarray
+        Affine transformation from voxels to streamlines. Default: identity.
```
Since this is something that pops up in almost all of our internal discussion on streamlines: should the affine documentation specify if the offset should be included in the transformation? By offset, I mean the same offset that is used when mapping streamlines to voxel coordinates to interact with a mask, which is normally 0.5 * the voxel size.

I would say that it is up to the user, and in most cases, it probably won't change the end result, but I'm curious what others would say.

jchoude commented:
Good, simple work. If there was an approve button, I would click it.

arokem commented:
Thanks for taking a look, @jchoude - and thanks for the comments. 

I added a couple more tests, to get full coverage, and I added a `copy` kwarg, so that streamlines can be copied or changed in-place (default is to copy).
